agent/profile

```diff
@@ -1,35 +1,44 @@
 # -*- mode: shell-script -*-
-prefix=/opt/pbench-agent
 
-if [[ ! -f ${prefix}/profile ]]
-then
-    if [[ -f ${PWD}/profile ]]
+function _pbench_pathins() {
+    if [ -d "$1" ] && [[ $PATH != *":$1:"* && $PATH != "$1"* && $PATH != *"$1" ]] ; then
+        PATH="$1${PATH:+:$PATH}"
+    fi
+}
+
+function _pbench_setup() {
+    local prefix=/opt/pbench-agent
+
+    if [[ ! -f ${prefix}/profile ]]
     then
-	prefix=$PWD
-    else
-	# last ditch attempt
-	prefix=$TOP
+	if [[ -f ${PWD}/profile ]]
+	then
+	    prefix=$PWD
+	else
+	    # last ditch attempt
+	    prefix=$TOP
+	fi
     fi
-fi
 
-pathins() {
-    if [ -d "$1" ] && [[ ":$PATH:" != *":$1:"* ]]; then
-        PATH="$1:$PATH"
+    if [[ ! -f ${prefix}/util-scripts/pbench-register-tool-set ]]
+    then
+	echo "*** WARNING *** Something wrong in the pbench base profile:" >&2
+	echo "                Setup has failed - could not determine the proper environment prefix: \"$prefix\"" >&2
+	echo "                You will need to fix it before working with pbench." >&2
+    else
+	local pdir=${prefix}/lib
+	if [[ -z "$PYTHONPATH" && $PYTHONPATH != *":$pdir:"* && $PYTHONPATH != "$pdir"* && $PYTHONPATH != *"$pdir" ]] ; then
+	    export PYTHONPATH=${prefix}/lib${PYTHONPATH:+:$PYTHONPATH}
+	fi
+	export _PBENCH_AGENT_CONFIG=${prefix}/config/pbench-agent.cfg
+	export PBENCH_ROOT=$prefix
```

Comment:
Unless we are going to use `PBENCH_ROOT` other places in the code and add it to the documentation in this commit, exporting it doesn't seem like a good idea.  We already have the `pbench_install_dir` enviroment variable created by `/opt/pbench-agent/base`.

Reply:
This is so external tools that need to know where pbench is installed can get at it.

Reply:
It is really worth having two ways to get the same information, given that it is well known that when multiple ways are given to get information invariably one method breaks over time?

We have been working to DRY our the code base, which is quite hard to do.  We should avoid adding to it if we can.

So perhaps we can consider documenting that external programs use a method like we find in [`base`]( https://github.com/distributed-system-analysis/pbench/blob/master/agent/base#L58), or change `base` to drop setting `pbench_install_dir` and instead expect it to be set via the `profile`.

Either way, for compatibility, let's use `pbench_install_dir` instead of `PBENCH_ROOT`.

Reply:
Any thoughts on this?  I don't think we should be adding this variable in light of what is already in `base`.

Reply:
Can you update this commit message and PR description to remove mention of the `PBENCH_ROOT` variable?  And open an issue for adding such an environment variable in the future?  I think it is a good idea, but I'd rather make the change along with the Python3 refactoring we are doing (see PR #1770, for example).

Reply:
Tracking this work in #1794